### ISSUE TITLE
Scalar Objects to PECL?

### ISSUE THREAD

**Issue by HallofFamer** [nikic/scalar_objects#27]
Hi there, I was wondering if it is possible to bring scalar objects extensions to PECL. I'd love to see scalar objects making it to PHP core one day, but in recent future its more realistic to bring it to PECL. Then more developers will start to install and use it, and once scalar objects grow in popularity, we can move on to the next stage. So what do you think @nikic ? Can you bring it to PECL first since it is qualified as an extension?

**Comment by nikic** [comment]
I'm fine with having this in PECL, as long as someone else takes care of the technicalities. I'm not capable of setting up a working PEAR/PECL environment. I'm pretty sure that it hates me, personally.

**Comment by HallofFamer** [comment]
I see, thats very good news. I will find a way to set up PECL environment, or to ask someone who already has PECL environment to host it. I will update you once it is done.

**Comment by HallofFamer** [comment]
Hi @nikic, I think its possible for me to set up a working PECL environment for Scalar Objects in your place. But still, you will have to publish it on PECL's website, by submitting this form. Will you do it? This is your extension after all, so there are some fields that I cannot fill in your place.

**Comment by HallofFamer** [comment]
@nikic 
Did you see my last comment? Is it possible for you to submit information regarding scalar objects as a package to PECL using its form? Its possible to have someone else including myself to set up the environment for PECL Scalar objects, but still its better for the creator to make PECL submission request.
https://pecl.php.net/account-request.php

**Comment by remicollet** [comment]
@nikic the README file states 

> The main purpose of this repo is to provide a **proof of concept** implementation

PoC are usually not design to be released ;)

BTW, if you plan to maintain this extension, and tag a release, I can take care of the pecl release process for you (as I already do for some others owner / extensions).

In this case, you only have to take care of announcing it on the pecl mailing list, and if no objection, ping me.

**Comment by HallofFamer** [comment]
@nikic 
What do you think about the proposal from Remi?

**Comment by HallofFamer** [comment]
@nikic its been almost 2 years, what is your current thought about this? Seems that @remicollet is willing to push this to PECL, or if not I can help you find someone else who can host Scalar Objects extension from PECL too.

**Comment by HallofFamer** [comment]
PHP 8 has been released, a long time has elapsed. Is there still no plan to bring scalar objects to PECL so far?

**Comment by Davidoc26** [comment]
So much time has passed, it's a pity that such a great opportunity that could have been in the language was forgotten ...

**Comment by HallofFamer** [comment]
I dont think it has been forgotten, and in fact the feature of scalar_objects is frequently brought up when developers discuss PHP features at /r/php subreddit. It seems that theres significant difficulty bringing this into PHP core, though I wonder if a similar feature like extension methods which allow both objects and scalar types to be 'extended' will be a better solution to this problem. 

Anyway, I really hope this one can at least, make it to PECL. I've contacted @nikic before but he seems uninterested, @remicollet even offered to host it incase he doesnt want to set up PECL account. I have no idea though, maybe hes just busy.
